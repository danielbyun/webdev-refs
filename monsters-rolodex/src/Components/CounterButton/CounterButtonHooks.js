import React, { useState, memo } from "react";

const CounterButtonHooks = (props) => {
  const [count, setCount] = useState(0);
  const { color } = props;

  return (
    <button id="counter" color={color} onClick={() => setCount(count + 1)}>
      Count: {count}
    </button>
  );
};

export default memo(CounterButtonHooks);
