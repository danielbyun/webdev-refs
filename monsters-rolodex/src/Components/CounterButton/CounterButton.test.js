import React from "react";

import { shallow } from "enzyme";
import CounterButton from "./CounterButton";
import CounterButtonHooks from "./CounterButtonHooks";

describe("CounterButton", () => {
  const mockColor = "black";
  const wrapper = shallow(<CounterButton color={mockColor} />);
  const hooksWrapper = shallow(<CounterButtonHooks color={mockColor} />);

  it("Expect to render CounterButton component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("Expect to render CounterButton Hooks Component", () => {
    expect(hooksWrapper).toMatchSnapshot();
  });

  it("Correctly increments the counter", () => {
    // simulates click event on the Id selected element
    wrapper.find("[id='counter']").simulate("click");
    hooksWrapper.find("[id='counter']").simulate("click");

    // grabs the state and compares
    expect(wrapper.state()).toEqual({ count: 1 });
    // expect(hooksWrapper.state("count")).toEqual({ count: 1 });

    expect(wrapper.props().color).toEqual("black");
  });
});
