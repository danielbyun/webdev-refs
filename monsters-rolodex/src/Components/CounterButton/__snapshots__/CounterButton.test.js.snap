// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterButton Expect to render CounterButton Hooks Component 1`] = `ShallowWrapper {}`;

exports[`CounterButton Expect to render CounterButton component 1`] = `ShallowWrapper {}`;
